backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/GetVmOvfByVmIdQuery.java

```diff
@@ -35,10 +39,16 @@ protected void executeQueryCommand() {
             return;
         }
 
+        // We want to lock the devices to make sure no other thread modifies them during the ovf export.
+        // Otherwise we end up without ovf in the export
+        resourceManager.getVmManager(getParameters().getId()).getVmDevicesLock().lock();
```

Comment:
that won't cover all cases - the db generation may change due to an operation (and not via the devices monitoring) and you also may get `vm.getDbGeneration() != getParameters().getRequiredGeneration()` above

Reply:
and also, we may get to this line with a vm that was queried with db-generation X from the database and right before getting here, the devices monitoring has already update the db-generation in the database to Y - so I think that this is not the way to go but even if someone is willing to block the monitoring thread like that, there need to be some changes here..